### PR TITLE
#159987658 Users should be able to like a specific comment

### PR DESCRIPTION
**Pull request by CEOehis**
#### What does this PR do?
Enable users to like/dislike a single comment

#### Description of Task to be completed?
- Track user reaction state on each comment
- update like/dislike count accordingly
- include relevant test

#### How should this be manually tested?
Navigate the app to `http:localhost:3000/articles/:articleSlug`
Then click `like` or `dislike` on any comment.

#### What are the relevant pivotal tracker stories?
[#159987658](https://www.pivotaltracker.com/story/show/159987658)

#### Screenshots (if appropriate)
screenshot 2018-10-17 at 9 42 55 pm

### CODE REVIEW

File: src/tests/actions/comments.test.js [hunk 1]
```diff
@@ -71,22 +75,41 @@ describe('comment actions', () => {
...
         {
-          type: types.CREATE_COMMENT_REQUEST,
-          payload: false
+          type: "CREATE_COMMENT_REQUEST",
+          payload: false,
         },
         {
-          type: types.CREATE_COMMENT_SUCCESS,
-          payload: {id: 1, body: 'The body'}
+          type: "CREATE_COMMENT_SUCCESS",
```
Comment (lauragift21) [inline src/tests/actions/comments.test.js]: You imported the actiontypes as * types from ../../actionTypes/comment but here you are calling it directly shouldn't it be  
```suggestion
          type: types.CREATE_COMMENT_SUCCESS,
```

**Review by lauragift21 — CHANGES_REQUESTED** [review]
LGTM

File: src/tests/actions/comments.test.js (hunk 1 above)
Comment (CEOehis) [reply src/tests/actions/comments.test.js]: Yes it should. Thanks for pointing that out. I'll fix it ASAP.

File: src/reducers/comments.js [hunk 2]
```diff
@@ -3,6 +3,39 @@ import * as types from '../actionTypes/comments';
...
 const { comments } = initialState;
 
+const commentsData = (state = [], action) => {
+  switch (action.type) {
+    case types.LIKE_COMMENT:
+      return state.map((comment, idx) => {
+        if (idx === Number(action.key)) {
+          return {
+            ...comment,
+            likesCount: comment.liked ? comment.likesCount - 1 : comment.likesCount + 1,
```
Comment (d-beloved) [inline src/reducers/comments.js:15]: Please add necessary comments here for easy comprehension of the logic

**Review by d-beloved — COMMENTED** [review]
LGTM

File: src/reducers/comments.js (hunk 2 above)
Comment (CEOehis) [reply src/reducers/comments.js:15]: Ok. I have added the relevant comments. Thanks.

**Review by d-beloved — APPROVED** [review]
LGTM

**Review by lauragift21 — APPROVED** [review]

**Review by johnkingzy — APPROVED** [review]
LGTM

**Pull request merged**